content/xc9_k8s_aks.md

```diff
@@ -0,0 +1,108 @@
++++
+title = "XC9 + K8S + AKS = Bingo?! 🌟"
+date = "2019-11-06"
+tags = ["Docker", "Azure", "XC9"]
++++
+
+Sitecore 9 has a micro-service architecture and is designed with the cloud in mind. The community has picked this up by creating Docker images for Sitecore. This works great for local development, but it still has some challenges when deploying to the cloud. This changed with the recent introduction of Windows support for Kubernetes (K8S) and Azure Kubernetes Service (AKS) support for Windows containers. 
+
+<!--more-->
+This article introduces a Kubernetes setup, using Sitecore XC 9.1 as showcase, that allows you to deploy to AKS.
+
+# Sitecore XC 9.1 topology
+Let's first have a look at the Sitecore XC 9.1 topology. 
+The below diagram gives an overview of all the services and databases for a basic XC 9.1 setup.
+
+![](/sitecore_xc91_topology.png)
+
+These services are mapped to several Docker images, this mapping is as follows;
+
+- Content Management & Content Delivery are combined in a **Sitecore** Docker image
+- Business Tools and the Commerce Engine roles are grouped in a **Commerce** Docker image
+- The Identity Server has its own Docker image, named **Identity**
+- XConnect and its Processing part are placed in a **XConnect** Docker image
+- All SQL database are present in a **Mssql** image
+- All Solr cores are placed in a **Solr** image
+
+These 6 Docker images form the basis for running the containerized Sitecore XC system.
+
+# Azure Kubernetes Service
+Now that we have a basic overview of the containerized setup, lets look at Azure Kubernetes Service (AKS).
+We run two types of nodes; a Linux and Windows node. The Linux node is used to run the Kubernetes services, and the Windows node is used for running the Sitecore Windows containers.
+A basic AKS cluster and its infrastructure is depicted below.
+
+![](/docker_deployment_variants.png)
+
+At the bottom we have the infrastructure for our cluster. It consists of a Linux and Windows VM, connected to eachother using a virtual network.
+
+On top of the infrastructure the Kubernetes cluster is created.
+The cluster consists of a master Linux node and a (slave) Windows node. Inside each node run a number of default services;
+
+- *kubelet*, which is the primary node agent
+- *k-proxy*, the network proxy that e.g. handles communication between nodes
+- *Docker* Moby as container runtime
+
+On the master node some additional services are running;
+
+- *api*, which is the Kubernetes server
+- *etcd*, this is the Kubernetes configuration store
+- *scheduler*, the service in charge of placing Pods in the cluster
+
+# Setup AKS cluster
+With this understanding of an AKS cluster its time to get hands-on and setup an empty hybrid (Linux & Windows nodes) Kubernetes cluster. 
+The movie below shows how to set it up. All performed steps are also described in detail [here](https://github.com/joostmeijles/xc9-k8s/blob/master/k8s/README.md).
+
+{{< youtube tZUqtDLOSs8 >}}
```

Comment:
Ik begrijp dat je de filmpjes wil hergebruiken. Ik wordt hier persoonlijk nooit blij van in een artikel. Ik zou dan juist naar een artikel omdat ik geen zin heb om een filmpje te bekijken. Het is dan best teleurstellend als de inhoud waar je naar opzoek bent alsnog in een filmpje staat. Ik zou ze zelf combineren. Dus in text beschrijven en een extra linkje naar het filmpje erbij.

Reply:
Eens. De beschrijving van de stappen is echter nogal uitgebreid, vandaar dat ik verwijs naar de verschillende READMEs in m'n Github repo. Zal het wat uitbreiden in het artikel en de filmpjes wat meer bijzaak maken.